tutorials/breakoutboard-tut.md

```diff
@@ -0,0 +1,90 @@
+
+### Analog outputs
+
+![Breakout board analog output processing graph](../images/breakout-tut-3.png)
+
+This portion of the workflow generates an ramping analog signal and sends it to each of the analog outputs using a <xref:OpenEphys.Onix1.BreakoutAnalogOutput> operator.  The `RampGenerator` is a [`GroupWorkflow`](https://bonsai-rx.org/docs/articles/editor.html#workflow) that contains an number of Bonsai operators. You can look at how it works hitting <kbd>Ctrl</kbd> + <kbd>Enter</kbd> when the operator is highlighted to expand the internal workflow in a new tab. The `RampGenerator` contains a ~100 Hz timer that is used to create an array of 12 ramping voltage values, one for each channel. Although a voltage ramp is sent to all the channels, only channel 0 was selected to be a output, so this is the only channel that will be affected. If other channels are configured as outputs (see [ConfigureBreakoutBoard](#configurebreakoutboard)), then they will also ramp their voltage.
+
+### Digital inputs
+
+![Breakout board digital input processing graph](../images/breakout-tut-5.png)
+
+The 8-bit, 5-V tolerant digital input port along with the breakout board's button and switch states are captured using a <xref:OpenEphys.Onix1.BreakoutDigitalInput> operator. The digital inputs are sampled at 5 MHz. However, they will only produce data when a change in state occurs (e.g. a digital input pin toggles from logic low to logic high or a button is pressed). Whenever a change in digital state occurs data will be propagated and saved by the `CsvWriter`. Because the `CsvWriter` is a [Sink](https://bonsai-rx.org/docs/articles/operators.html#sink), the <xref:OpenEphys.Onix1.BreakoutDigitalInputDataFrame> produced by the <xref:OpenEphys.Onix1.BreakoutDigitalInput> operator will pass right through it and can be expanded by right-click the operator and hovering over `Output` in the context menu. In this case we have exposed `DigitalInputs` and `Buttons`, which represent the 8-bit digital input port state and the button/switch state, respectively. Each of these outputs is followed by a `HasFlag` filter that is used to determine if a certain digital input pin is logic-high (Pins 0, 1, and 7 in this example) or to check if a certain button (☾ or □ in this example) has been depressed.
+
+### Digital outputs
+
+![Breakout board digital output processing graph](../images/breakout-tut-6.png)
+
+The 8-bit digital output port is updated using a <xref:OpenEphys.Onix1.BreakoutDigitalOutput> operator. A ~10 Hz timer is used to drive a counter. Although this produced a 32-bit integer that counts from 1 to 2147483647,  <xref:OpenEphys.Onix1.BreakoutDigitalOutput> operator will only use the lower 8-bits to update the digital output state. This can be seen reflected on the LEDs on the breakout board will show a binary count for 0 to 255 before the pattern repeats.
```

Comment:
Thanks for the feedback:

- SVG - yeah for sure. Also a non numerical file naming scheme because numerical ordering makes it weird to put pictures in between existing ones
- Wall of text - I'm hoping that the section is further expanded with mor explanatory images. I would like you to look into that when you have time.

So, I will go through these edits and implement. Also I want to make a data loading script, which I will put at the bottom. After that, it would be good if you could flesh out with even more images.